workflow/controller/operator.go

```diff
@@ -2977,6 +2978,26 @@ func (woc *wfOperationCtx) executeSuspend(nodeName string, templateScope string,
 	return node, nil
 }
 
+func resolveInputFieldsForSuspendTemplate(tmpl *wfv1.Template) {
+	parameters := tmpl.Inputs.Parameters
+	for i, parameter := range parameters {
+		if parameter.Value != nil {
+
+			value := parameter.Value.String()
+			valueMap := map[string][]wfv1.AnyString{}
```

Comment:
I think you should create a struct for this

Reply:
Reused Parameter